extension/src/plots/model/collect.ts

```diff
@@ -80,19 +80,65 @@ const getValues = (
   return values
 }
 
+const filterColorScale = (
+  colorScale: ColorScale | undefined,
+  valueIds: string[]
+) => {
+  const filteredColorScale: ColorScale = {
+    domain: [],
+    range: []
+  }
+
+  if (!colorScale) {
+    return filteredColorScale
+  }
+
+  for (const [ind, id] of colorScale.domain.entries()) {
+    if (!valueIds.includes(id)) {
+      continue
+    }
+    filteredColorScale.domain.push(id)
+    filteredColorScale.range.push(colorScale.range[ind])
+  }
+
+  return filteredColorScale
+}
+
+const fillColorScale = (
+  experiments: Experiment[],
+  colorScale: ColorScale | undefined,
+  valueIds: string[]
+) => {
+  const completeColorScale = filterColorScale(colorScale, valueIds)
```

Comment:
We need to filter through the original color scale values first so they retain their order in the legend. Maybe there's a simpler way of doing this though 🤔

Reply:
something like 
```
const removeSelectedExperiment = (
  orderedColorScale: ColorScale,
  hasValues: boolean,
  idx: number
) => {
  const isSelectedExperiment = idx !== -1
  if (!isSelectedExperiment || hasValues) {
    return
  }

  orderedColorScale.domain.splice(idx, 1)
  orderedColorScale.range.splice(idx, 1)
}

const fillColorScale = (
  experiments: Experiment[],
  colorScale: ColorScale | undefined,
  valueIds: string[]
) => {
  const orderedColorScale = {
    domain: [...(colorScale?.domain || [])],
    range: [...(colorScale?.range || [])]
  }

  for (const experiment of experiments) {
    const { id } = experiment
    const idx = orderedColorScale.domain.indexOf(id)
    const isSelectedExperiment = idx !== -1
    const hasValues = valueIds.includes(id)

    if (!hasValues || isSelectedExperiment) {
      removeSelectedExperiment(orderedColorScale, hasValues, idx)
      continue
    }

    orderedColorScale.domain.push(id)
    orderedColorScale.range.push('#4c78a8' as Color)
  }

  return orderedColorScale
}
```

Reply:
Works great, thanks!